src/plugins/intel_gpu/src/graph/layout_optimizer.cpp

```diff
@@ -1866,11 +1873,13 @@ void layout_optimizer::select_preferred_formats_for_onednn(program_node& node, d
                     dst_fmt = format::b_fs_yx_fsv32;
                 }
             }
+            auto forced_format = get_forced_format(node);
+            if (forced_format != format::any)
+                dst_fmt = forced_format;
 
-            if (node.get_preferred_output_fmt() == format::any) {
-                for (size_t usr = 0; usr < std::max<size_t>(1, node.get_users().size()); usr++)
+            for (size_t usr = 0; usr < std::max<size_t>(1, node.get_users().size()); usr++)
+                if (node.get_preferred_output_fmt(usr) == format::any)
                     node.set_preferred_output_fmt(usr, dst_fmt);
```

Comment:
It is related to multi-output port. I don't think it is correct to fix this way.